### DISCUSSION TITLE
is a closed database safe for file copy?

### DISCUSSION THREAD

**Discussion by mattsimgithub** [simolus3/drift#1462] [Q&A] [answered]
I want to do something like this:

```dart
  final db = Provider.of<Database>(context, listen: false);
  await db.close();

  await _copyDatabaseFileToBackupFile();

  await _uploadBackupFile();

  _deleteBackupFile();

```

`db` is actually a LazyDatabase, if this is important.

From what else I've seen, I infer this is OK as long as `db.close()` is properly awaited. But I'd like to doublecheck.

Also, is it possible to re-open the database easily once it's closed (like, after all the other operations are complete)?

**Comment by simolus3** [comment, marked as answer]
> From what else I've seen, I infer this is OK as long as `db.close()` is properly awaited. But I'd like to doublecheck.

Yes, this should be fine. Be aware that other operations using the database will throw an exception once the database has been closed, so you might need some in-app synchronization to avoid that.

> Also, is it possible to re-open the database easily once it's closed

Not really unfortunately, you'd have to create a new instance of the database class and use that then.

**Reply by mattsimgithub** [reply]
Everything you say makes sense. Thanks for the reply.
